hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/HapiTestEngine.java

```diff
@@ -138,19 +140,34 @@ public String getId() {
     @Override
     public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId uniqueId) {
         final var engineDescriptor = new EngineDescriptor(uniqueId, "Hapi Test");
+
+        // We obtain the selectors using MethodSelector and ClassSelector because we don't know how the
+        // tests will be executed. If we run tests at the class level, we will have a ClassSelector.
+        // If we run a single test, we will have a MethodSelector.
```

Comment:
If this is true and we want to proceed with this path we should also add support for `ClasspathRootSelector` so we can run all tests at a package level.

I can add that in this PR but I need confirmation that it's needed because I think it would be a bit more complicated than the rest of selectors.

Reply:
I think this makes sense. It was @rbair23's original idea. Let's wait for his thoughts. If this is more complicated, let's do it in a second PR and get this one merged independently.